src/mockServiceWorker.js

```diff
@@ -82,22 +82,26 @@ self.addEventListener('message', async function (event) {
   }
 })
 
-async function resolveClient(event) {
+async function resolveMasterClient(event) {
   const client = await self.clients.get(event.clientId)
+
+  if (client.frameType === 'top-level') {
+    return client
+  }
+
   const allClients = await self.clients.matchAll()
-  return client.frameType === 'top-level'
-    ? client
-    : allClients.find((parentClient) => {
-        return (
-          parentClient.frameType === 'top-level' &&
-          parentClient.visibilityState === 'visible' &&
-          client.url.startsWith(parentClient.url)
-        )
-      })
+
+  return allClients
+    .filter((client) => {
+      return client.visibilityState === 'visible'
+    })
+    .find((client) => {
+      return activeClientIds.has(client.id)
```

Comment:
No matter the number of nested clients (i.e. iframes), one of them is the one that registered the worker (master client). We can look up if its ID is in the stored active clients' Map and resolve if it is, otherwise, it means that the master client hasn't sent the activation message to the worker yet and the requests must be bypassed.